target/linux/ramips/dts/mt7620a_tplink_re200-v1.dts

```diff
@@ -0,0 +1,158 @@
+/dts-v1/;
+
+#include "mt7620a.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "tplink,re200-v1", "ralink,mt7620a-soc";
+	model = "TP-Link RE200 v1";
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,57600n8";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		
+		led_power: power {
+		    label = "re200-v1:green:power";
+		    gpios = <&gpio1 0 GPIO_ACTIVE_LOW>;
+		};
+		
+		lan {
+			label = "re200-v1:green:lan";
+			gpios = <&gpio2 0 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan {
+			label = "re200-v1:green:wlan";
+			gpios = <&gpio1 3 GPIO_ACTIVE_LOW>;
+		};
+		
+		qss {
+		    label = "re200-v1:green:qss";
+		    gpios = <&gpio1 15 GPIO_ACTIVE_LOW>;
+		};
+		
+		led_2g1 {
+		    label = "re200-v1:red:2g";
+		    gpios = <&gpio0 2 GPIO_ACTIVE_LOW>;
+		};
+		
+		led_2g2 {
+		    label = "re200-v1:green:2g";
+		    gpios = <&gpio3 0 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 13 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio0 12 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+};
+
+
+&state_default {
+	gpio {
+	    ralink,group = "i2c", "uartf", "ephy", "wled", "nd_sd", "rgmii1", "spi refclk";
+		ralink,function = "gpio";
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	m25p80@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x20000>;
+				read-only;
+			};
+
+			partition@20000 {
+				compatible = "tplink,firmware";
+				label = "firmware";
+				reg = <0x20000 0x7c0000>;
+			};
+
+			userconfig: partition@7e0000 {
+				label = "userconfig";
+				reg = <0x7e0000 0x10000>;
+				read-only;
+			};
+
+			radio: partition@7f0000 {
+				label = "radio";
+				reg = <0x7f0000 0x10000>;
+				read-only;
+			};
+		};
+	};
+};
+
+&gpio0 {
+    status = "okay";
+};
+
+&gpio1 {
+	status = "okay";
+};
+
+&gpio2 {
+	status = "okay";
+};
+
+&gpio3 {
+	status = "okay";
+};
+
+&ethernet {
+	mtd-mac-address = <&userconfig 0x0158>;
+};
+
+&wmac {
+	ralink,mtd-eeprom = <&radio 0>;
```

Comment:
Is there a MAC address in radio 0x4 and/or 0x8004?

Reply:
Hard to tell, definitely not the one written at the backside. The MACs I'm using are more or less consistent with original firmware (they assign the same MAC to one of the radios and the LAN device, but they skip the MAC between the two).

Reply:
Have a look at: https://openwrt.org/docs/guide-developer/mac.address

Reply:
I searched the whole flash for the MAC address printed at the bottom of the device and I could only find it in the userconfig partition at the locations present in the dts. 
There are three addresses:
*:88
*:89
*:89

When running stock firmware, my device assigns *:88 to LAN and 2.4G WiFi and *:8A to the 5G WiFi. Thus, I use *:88 for LAN, *:89 for 2.4G and *:8A for 5G WiFi.

Reply:
So, you have the following?:
```
MAC assignment based on stock firmware:
*:88    Ethernet/2.4G    userconfig 0x0158
*:89    unused         userconfig 0x0160
*:8A    5G                 not present in flash
```

If it is correct, please add this to your commit message (and maybe format it more nicely.

Please stick to the MAC address assignment of the stock firmware. You can never be sure whether the additional unused MAC address will be there on other devices. For the same reason, I would calculated 5G Wifi based on 0x158, so adding +2 instead of +1.

Normally, userconfig partition is for stuff created by OpenWrt firmware. So, I would expect to find one of the MAC address in u-boot, too, and would base everything on that one (if it really exists).

Which of the addresses is on the label?

Reply:
Correct, the *:88 is on the label. 
Yes, the *:88 is also in the U-Boot partition at 0x1fc00 - seems to be a safe guess then. I changed it as you suggested.

Reply:
Perfect. That would be the standard TP-Link position I expected.